gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/infra/domain_service/subscription/SubscriptionSpecDomainServiceImpl.java

```diff
@@ -0,0 +1,113 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.gravitee.apim.infra.domain_service.subscription;
+
+import io.gravitee.apim.core.audit.model.AuditInfo;
+import io.gravitee.apim.core.subscription.domain_service.AcceptSubscriptionDomainService;
+import io.gravitee.apim.core.subscription.domain_service.CloseSubscriptionDomainService;
+import io.gravitee.apim.core.subscription.domain_service.SubscriptionSpecDomainService;
+import io.gravitee.apim.core.subscription.model.SubscriptionEntity;
+import io.gravitee.apim.core.subscription.model.crd.SubscriptionSpec;
+import io.gravitee.apim.infra.adapter.SubscriptionAdapter;
+import io.gravitee.rest.api.service.SubscriptionService;
+import io.gravitee.rest.api.service.common.ExecutionContext;
+import io.gravitee.rest.api.service.exceptions.SubscriptionNotFoundException;
+import java.time.ZonedDateTime;
+import java.util.Objects;
+import java.util.Optional;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+
+/**
+ * @author Antoine CORDIER (antoine.cordier at graviteesource.com)
+ * @author GraviteeSource Team
+ */
+@Slf4j
+@Service
+@RequiredArgsConstructor
+public class SubscriptionSpecDomainServiceImpl implements SubscriptionSpecDomainService {
+
+    private static final String ACCEPT_REASON = "Kubernetes subscriptions are always auto accepted";
+
+    private final SubscriptionService subscriptionService;
+
+    private final SubscriptionAdapter adapter;
+
+    private final AcceptSubscriptionDomainService acceptService;
+
+    private final CloseSubscriptionDomainService closeSubscriptionDomainService;
+
+    @Override
+    public SubscriptionEntity createOrUpdate(AuditInfo auditInfo, SubscriptionSpec spec) {
+        return find(spec.getId()).map((existing -> update(auditInfo, existing, spec))).orElseGet(() -> create(auditInfo, spec));
+    }
+
+    @Override
+    public void delete(AuditInfo auditInfo, String id) {
+        closeSubscriptionDomainService.closeSubscription(id, auditInfo);
+    }
+
+    private SubscriptionEntity create(AuditInfo auditInfo, SubscriptionSpec spec) {
+        var entity = adapter.fromSpec(spec);
+
+        var then = ZonedDateTime.now();
+
+        entity.setEnvironmentId(auditInfo.environmentId());
+        entity.setSubscribedBy(auditInfo.actor().userId());
+        entity.setProcessedBy(auditInfo.actor().userId());
+        entity.setCreatedAt(then);
+        entity.setProcessedAt(then);
+        entity.setStartingAt(then);
+        entity.setReasonMessage(ACCEPT_REASON);
+        entity.setEndingAt(spec.getEndingAt());
+
+        var subscription = subscriptionService.create(toExecutionContext(auditInfo), adapter.fromCoreForCreate(entity), null, spec.getId());
+
+        if (entity.getEndingAt() != null) {
```

Comment:
Didn't we set this value already on line 76?

Reply:
It's because unfortunately the create does not set this field, so we need to update if the ending date is set on create